core/includes/install.core.inc

```diff
@@ -1454,6 +1450,372 @@ function install_select_language_form($form, &$form_state, $files) {
+
+  // Then, attempt to create or make writable the translations directory.
+  file_prepare_directory($translations_directory, FILE_CREATE_DIRECTORY | FILE_MODIFY_PERMISSIONS);
+
+  // Get values so the requirements errors can be specific.
+  if (backdrop_verify_install_file($translations_directory, FILE_EXIST, 'dir')) {
+    $readable = is_readable($translations_directory);
+    $writable = is_writable($translations_directory);
+    $translations_directory_exists = TRUE;
+  }
+
+  $version = BACKDROP_VERSION;
+
+  // For dev releases, remove the '-dev' part and trust the translation server
+  // to fall back to the latest stable release for that branch.
+  // @see locale_translation_build_projects()
+  if (preg_match("/^(\d+\.).*-(dev|preview)$/", $version, $matches)) {
+    // Example match: 1.28-dev => 1.x (Backdrop core).
+    $version = $matches[1] . 'x';
+  }
+
+  // Build URL for the translation file and the translation server.
+  $variables = array(
+    '%project' => 'backdropcms',
+    '%version' => $version,
+    '%core' => 'all',
+    '%language' => $langcode,
+  );
+  $translation_url = strtr($server_pattern, $variables);
+
+  $elements = parse_url($translation_url);
+  $server_url = $elements['scheme'] . '://' . $elements['host'];
+
+  // Build the language name for display.
+  include_once BACKDROP_ROOT . '/core/includes/standard.inc';
+
+  $languages = standard_language_list();
+  $language = isset($languages[$langcode]) ? $languages[$langcode][0] : $langcode;
+
+  // Check if any of the desired translation files are available or if the
+  // translation server can be reached. In other words, check if we are online
+  // and have an internet connection.
+  if (install_check_localization_server($server_url)) {
+    $online = TRUE;
+  }
+  if ($online) {
+    $translation_available = install_check_localization_server($translation_url);
+  }
+
+  // If the translations directory does not exist, throw an error.
+  if (!$translations_directory_exists) {
+    $requirements['translations directory exists'] = array(
+      'title'       => st('Translations directory'),
+      'value'       => st('The translations directory does not exist.'),
+      'severity'    => REQUIREMENT_ERROR,
+      'description' => st('The installation process requires a translations directory to be created under <code>%translations_directory</code>. The <a href="!install_docs">How to install Backdrop CMS</a> documentation page offers help on this and other topics.', array(
+        '%translations_directory' => $translations_directory,
+        '!install_docs' => 'https://docs.backdropcms.org/documentation/installation-instructions',
+      )),
+    );
+  }
+  else {
+    $requirements['translations directory exists'] = array(
+      'title'       => st('Translations directory'),
+      'value'       => st('The directory <code>%translations_directory</code> exists.', ['%translations_directory' => $translations_directory]),
```

Comment:
We still have a mix of long and short array syntax in this file.

Short array here (st() params) and also in line 1727 and 1738 and 1749. I don't have strong feelings re that, but mixing both seems a bit confusing.

Reply:
Resolved